src/pyuvdata/utils/tools.py

```diff
@@ -584,3 +584,98 @@ def _where_combine(mask, inds=None, invert=False, use_and=True):
         mask = eval_func(mask, postmask)
 
     return np.nonzero(np.logical_not(mask) if invert else mask)[0]
+
+
+def _nants_to_nblts(uvd):
+    """
+    Obtain indices to convert (Nants,) to (Nblts,).
+
+    Parameters
+    ----------
+    uvd : UVData object
+
+    Returns
+    -------
+    ind1, ind2 : ndarray, ndarray
+        index pairs to compose (Nblts,) shaped arrays for each
+        baseline from an (Nants,) shaped array
+    """
+    ants = uvd.telescope.antenna_numbers
+
+    ant1 = uvd.ant_1_array
+    ant2 = uvd.ant_2_array
+
+    ind1 = []
+    ind2 = []
+
+    for i in ant1:
+        ind1.append(np.where(ants == i)[0][0])
+    for i in ant2:
+        ind2.append(np.where(ants == i)[0][0])
```

Comment:
One other thing (that I thought I put in my review but maybe I accidentally deleted?) -- there are a lot of calls to `np.where` and `list.append` here, which can be fairly slow, especially when iterating through a long list like the above. Just a suggestion, you might think of using something like a dictionary to speed up (by more than an order of magnitude -- ~30x with the code below) mapping antenna number to index position, e.g.:

```suggestion
    ant_map = {ant: idx for idx, ant in enumerate(uvd.telescope.antenna_numbers)}
    
    ind1 = [ant_map[ant] for ant in uvd.ant_1_array]
    ind2 = [ant_map[ant] for ant in uvd.ant_2_array]
```